lupa/tests/test.py

```diff
@@ -2619,6 +2619,119 @@ def test_nil_debug_traceback(self):
         except lupa.LuaError as e:
             self.assertNotIn("stack traceback:", e.args[0])
 
+################################################################################
+# tests for handling overflow
+
+class TestOverflow(SetupLuaRuntimeMixin):
+    maxinteger = sys.maxsize # maximum value for C Py_ssize_t
+    biginteger = (maxinteger + 1) << 1 # value too big to fit in C size_t
+    maxfloat = sys.float_info.max # maximum value for Python float
+    bigfloat = int(maxfloat) * 2 # value too big to fit in Python float
+
+    assert biginteger <= maxfloat
+
+    def setUp(self):
+        super(TestOverflow, self).setUp()
+        self.lua_type = self.lua.eval('type')
+        self.lua_math_type = self.lua.eval('math.type')
+
+    def tearDown(self):
+        self.lua_type = None
+        self.lua_math_type = None
+        super(TestOverflow, self).tearDown()
+
+    def test_no_overflow(self):
+        self.assertMathType(0, 'integer')
+        self.assertMathType(10, 'integer')
+        self.assertMathType(-10, 'integer')
+        self.assertMathType(self.maxinteger, 'integer')
+        self.assertMathType(-self.maxinteger, 'integer')
+        self.assertMathType(0.0, 'float')
+        self.assertMathType(-0.0, 'float')
+        self.assertMathType(10.0, 'float')
+        self.assertMathType(-10.0, 'float')
+        self.assertMathType(3.14, 'float')
+        self.assertMathType(-3.14, 'float')
+        self.assertMathType(self.maxfloat, 'float')
+        self.assertMathType(-self.maxfloat, 'float')
+
+    def assertMathType(self, number, math_type):
+        self.assertEqual(self.lua_type(number), 'number')
+        if self.lua_math_type is not None:
+            self.assertEqual(self.lua_math_type(number), math_type)
+
+class TestOverflowWithoutHandler(TestOverflow, unittest.TestCase):
+    lua_runtime_kwargs = dict(overflow_handler=None)
+
+    def test_overflow(self):
+        self.assertRaises(OverflowError, self.assertMathType, self.biginteger, 'integer')
+        self.assertRaises(OverflowError, self.assertMathType, int(self.maxfloat), 'integer')
+        self.assertRaises(OverflowError, self.assertMathType, self.bigfloat, 'integer')
+
+class TestOverflowWithFloatHandler(TestOverflow, unittest.TestCase):
+    lua_runtime_kwargs = dict(overflow_handler=float)
+
+    def test_overflow(self):
+        self.assertMathType(self.biginteger, 'float')
+        self.assertMathType(int(self.maxfloat), 'float')
+        self.assertRaises(OverflowError, self.assertMathType, self.bigfloat, 'float')
+
+class TestOverflowWithObjectHandler(TestOverflow, unittest.TestCase):
+
+    def test_overflow(self):
+        self.lua.execute('python.set_overflow_handler(function(o) return o end)')
+        self.assertEqual(self.lua.eval('type')(int(self.maxfloat)), 'userdata')
+
+class TestBadOverflowHandlerInPython(unittest.TestCase):
+    def test_error(self):
+        self.assertRaises(ValueError, lupa.LuaRuntime, overflow_handler=123)
```

Comment:
This suggests to me that it should in fact raise a `TypeError`.

Reply:
But I do the same checking as for what is done with `attribute_filter`. I just check if it is `None` or callable.

Reply:
Hmm. I guess that was a bad decision back then as well. Not sure if it matters enough to introduce an inconsistency, though. Let's stick with what's there.